src/main/java/frc/robot/subsystems/RollerSubsystem.java

```diff
@@ -129,34 +132,24 @@ public void setRollPower(double power) {
 
     @Override
     public void periodic() {
-        boolean opening = openTimer.get() > 0 && !openTimer.hasElapsed(OPEN_TIME_SECONDS);
-        boolean closing = !opening || !allowOpen;
-
-        if (closing) {
+        // If we're opening and not allowed to anymore, or if we've finished opening, stop opening and start closing.
+        if ((openTimer.hasStarted() && !allowOpen) || openTimer.hasElapsed(OPEN_TIME_SECONDS)) {
+            openTimer.stop();
+            openTimer.reset();
             closeTimer.start();
+            cooldownTimer.start();
         }
 
-        if (closeTimer.get() > 0) {
-            if (!closeTimer.hasElapsed(CLOSE_TIME_SECONDS)) {
-                openMotor.set(-.2);
-            } else {
-                openMotor.set(0);
-            }
-        } else {
-            openMotor.set(0.5);
+        // If we're closing and finished, stop closing.
+        if (closeTimer.hasElapsed(CLOSE_TIME_SECONDS)) {
+            closeTimer.stop();
+            closeTimer.reset();
         }
 
-        // If `OPEN_TIME` hasn't elapsed yet, run the motor.
-        // boolean opening = openTimer.get() > 0 && !openTimer.hasElapsed(OPEN_TIME_SECONDS);
-        // if (opening && allowOpen) {
-        //     openMotor.set(opening && allowOpen ? 0.5 : 0);
-        // }
-
-        // If the cooldown has passed, stop and reset the timer.
-        if (openTimer.hasElapsed(OPEN_TIME_SECONDS + COOLDOWN_SECONDS)) {
-            openTimer.stop();
-            openTimer.reset();
-        }
+        // Otherwise, open if we're opening and close if we're closing.
+        if (openTimer.hasStarted()) openMotor.set(0.5);
+        else if (closeTimer.hasStarted()) openMotor.set(-0.2);
+        else openMotor.set(0);
```

Comment:
it might be tricky to integrate this with piece detection & powered clamping, if we use the latter solution. the idea would be to only use the motor to clamp actively (e.g. apply low voltage) if we're holding a cone, otherwise use 0 power.